plugins/nodes/src/js/components/NodesTable-new.tsx

```diff
@@ -0,0 +1,311 @@
+
+import {
+  hostnameSorter,
+  hostnameRenderer,
+  hostnameSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableHostnameColumn";
+import {
+  regionSorter,
+  regionRenderer,
+  regionSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableRegionColumn";
+import {
+  zoneSorter,
+  zoneRenderer,
+  zoneSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableZoneColumn";
+import {
+  healthSorter,
+  healthRenderer,
+  healthSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableHealthColumn";
+import {
+  tasksSorter,
+  tasksRenderer,
+  tasksSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableTasksColumn";
+import {
+  cpubarRenderer,
+  cpubarSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableCPUBarColumn";
+import {
+  cpuSorter,
+  cpuRenderer,
+  cpuSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableCPUColumn";
+import {
+  membarRenderer,
+  membarSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableMemBarColumn";
+import {
+  memSorter,
+  memRenderer,
+  memSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableMemColumn";
+import {
+  diskbarRenderer,
+  diskbarSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableDiskBarColumn";
+import {
+  diskSorter,
+  diskRenderer,
+  diskSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableDiskColumn";
+import {
+  gpubarSizer,
+  gpubarRenderer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableGPUBarColumn";
+import {
+  gpuSorter,
+  gpuRenderer,
+  gpuSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableGPUColumn";
+
+interface NodesTableProps {
+  hosts: NodesList;
+  nodeHealthResponse: boolean;
+  masterRegion: string;
+}
+
+interface NodesTableState {
+  data: Node[];
+  sortDirection: SortDirection;
+  sortColumn: string;
+}
+
+// This whole component (except `render()`) will basically go into data layer.
+// After we moved to DL, `NodesTable` component should be stateless/functional.
+export default class NodesTable extends React.Component<
+  NodesTableProps,
+  NodesTableState
+> {
+  constructor() {
+    super();
+
+    this.state = {
+      data: [],
+      sortColumn: "hostname",
+      sortDirection: "ASC"
+    };
+
+    this.handleSortClick = this.handleSortClick.bind(this);
+  }
+
+  sortData(data: Node[], sortColumn: string): Node[] {
+    switch (sortColumn) {
+      case "hostname":
+        return hostnameSorter(data);
+      case "region":
+        return regionSorter(data);
+      case "zone":
+        return zoneSorter(data);
+      case "health":
+        return healthSorter(data);
+      case "tasks":
+        return tasksSorter(data);
+      case "cpu":
+        return cpuSorter(data);
+      case "mem":
+        return memSorter(data);
+      case "disk":
+        return diskSorter(data);
+      case "gpu":
+        return gpuSorter(data);
+      default:
+        return data;
+    }
+  }
+
+  handleSortClick(columnName: string): void {
+    if (columnName === this.state.sortColumn) {
```

Comment:
🤔Is there a special reason why we don't have this state in our URL? Could be a nice thing for a customer to share a page with it's sorting. And we could make this component pretty dumb (besides holding the sorted data)

Reply:
it wasnt there before 🤷‍♂️ I agree that it would be nice, we should keep that in mind when introducing DL here 👍

Reply:
👍